tests/core/form/test_abstract_form.py

```diff
@@ -0,0 +1,181 @@
+import pytest
+
+from cascade.core.form.abstract_form import Form, Field, SimpleTypeField
+
+
+def test_development_target():
+    class IntField(Field):
+        def _validate_and_normalize(self, instance, value):
+            try:
+                new_value = int(value)
+            except (ValueError, TypeError):
+                return None, f"Invalid integer value '{value}'"
+            return new_value, None
+
+    class MyInnerForm(Form):
+        my_inner_field = IntField()
+
+    class MyForm(Form):
+        my_field = IntField()
+        my_inner_form = MyInnerForm()
+
+    f = MyForm({"my_field": "10", "my_inner_form": {"my_inner_field": 100}})
+    errors = f.validate_and_normalize()
+    assert not errors
+
+    assert f.my_field == 10
+    assert f.my_inner_form.my_inner_field == 100
+
+    f = MyForm({"my_field": "10", "my_inner_form": {"my_inner_field": "eaoeao"}})
+    errors = f.validate_and_normalize()
+    assert set(errors) == {("my_inner_form.my_inner_field", "Invalid integer value 'eaoeao'")}
+
+    f = MyForm({"my_field": None, "my_inner_form": {}})
+    errors = f.validate_and_normalize()
+    assert set(errors) == {
```

Comment:
I like that it returns multiple errors. I know that's what we need. It's just nice.